network/src/transport.rs

```diff
@@ -132,56 +143,56 @@ pub struct Connection<TSocket> {
     pub metadata: ConnectionMetadata,
 }
 
-fn identity_key_to_peer_id(
-    trusted_peers: &RwLock<HashMap<PeerId, NetworkPublicKeys>>,
-    remote_static_key: &[u8],
-) -> Option<PeerId> {
-    for (peer_id, public_keys) in trusted_peers.read().unwrap().iter() {
-        if public_keys.identity_public_key.to_bytes() == remote_static_key {
-            return Some(*peer_id);
-        }
+/// Convert a remote pubkey into a network `PeerId`.
+fn identity_pubkey_to_peer_id(
```

Comment:
the remote peer should submit a peer id in noise, that should either match their provided public key or be indexed into trusted peers... this will mitigate id confusion which I fear is happening right now in some of our tests

I left a comment here: https://github.com/libra/libra/issues/3960#issuecomment-637250990

and spoke with @mimoo to nail this down...

Reply:
for what it is worth this doesn't violate the proposal it just isn't as strict :)

Reply:
> the remote peer should submit a peer id in noise

you mean send it as part of the handshake?